stlab/channel.hpp

```diff
@@ -352,17 +388,24 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
     }
 
     void clear_to_send() override {
+        // The downstream sender may trigger this because an other parallel upstream
+        // process may have just passed down a value
+        if (_process_suspend_count == 0)
+            return;
+
         bool do_run = false;
         {
             std::unique_lock<std::mutex> lock(_process_mutex);
+            auto ps = get_process_state(_process);
```

Comment:
This is a race on get_process_state() - we have to wait until we know we are no longer suspended to ensure that it is only be entered by one thread.